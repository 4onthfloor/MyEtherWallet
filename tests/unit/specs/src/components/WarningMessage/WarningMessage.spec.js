import Vue from 'vue';
import WarningMessage from '@/components/WarningMessage/WarningMessage.vue';
import { shallowMount } from '@vue/test-utils';
import { Tooling } from '@@/helpers';

describe('WarningMessage.vue', () => {
  let localVue, i18n, wrapper, store;

  beforeAll(() => {
    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = baseSetup.store;
    Vue.config.warnHandler = () => {};
  });

  beforeEach(() => {
    wrapper = shallowMount(WarningMessage, {
      localVue,
      i18n,
      store,
      attachToDocument: true,
      propsData: {
        warningType: 'access'
      }
    });
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should dismount properly', () => {
    expect(wrapper.destroy()).toBe(undefined);
  });
});
