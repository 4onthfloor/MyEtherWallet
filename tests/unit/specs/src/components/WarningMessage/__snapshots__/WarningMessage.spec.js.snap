// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningMessage.vue renders correctly 1`] = `
<div class="warning-message">
  <div class="left">⚠️</div>
  <div class="right">
    <p class="title">Not recommended</p>
    <p class="message">
      This is not a recommended way to access your wallet.
      Due to the sensitivity of the information involved, these options should only be used in offline settings by experienced users.
    </p>
    <div class="link">
      &gt; Read:
      <a href="https://kb.myetherwallet.com/posts/offline/using-mew-offline/" target="_blank" rel="noopener noreferrer">Using MEW Offline</a></div>
  </div>
</div>
`;
